admin/services/check-window-v2.service.js

```diff
@@ -25,8 +25,17 @@ checkWindowV2Service.getCheckWindow = async (urlSlug) => {
  * Get active check window
  * @return {Promise<Object>} Check window object
  */
-checkWindowV2Service.getActiveCheckWindow = async () => {
-  return checkWindowDataService.sqlFindActiveCheckWindow()
+let cachedActiveCheckWindow
+let cachedActiveCheckWindowExpiresAt
+checkWindowV2Service.getActiveCheckWindow = async (cacheBust = false) => {
+  const now = (new Date()).getTime()
+
+  if (cacheBust || !cachedActiveCheckWindow || !cachedActiveCheckWindowExpiresAt || now > cachedActiveCheckWindowExpiresAt) {
+    cachedActiveCheckWindow = await checkWindowDataService.sqlFindActiveCheckWindow()
+    cachedActiveCheckWindowExpiresAt = (new Date()).getTime() + 60 * 1000 // +60 seconds
```

Comment:
why not use redis TTL and pass it to `redisService.set` or am i misunderstanding this?

Reply:
The issue, for me at least, was discovering that an express middleware function is called on every request which makes a network call to get the current active check window.  This modification stores it in memory for 60 seconds reducing network traffic.  It requires a trivial amount of memory, and the data is very slow-moving.  

After 60 seconds it will refresh itself from Redis (unfortunately the Redis cache implementation is at the data service level, so it is not clear.)

If the Redis cache is missing it will read from the Database.